lib/datadog/appsec/processor.rb

```diff
@@ -39,6 +39,27 @@ def finalize
         end
       end
 
+      class InvalidContextError < StandardError; end
+
+      class << self
+        def current_context
+          Thread.current[:datadog_current_waf_context]
+        end
+
+        def current_context=(context)
```

Comment:
It feels like this should be private.

Reply:
In that sense, do you see setting the active context as part of calling `new_context`? 

```ruby
def new_context
  context = Context.new(self)
  self.class.current_context = context
end
```

What is the intent of making this method private?  I'm not sure having a bunch of `send(:method_name)` in the codebase makes sense